calico_containers/calico_ctl/container.py

```diff
@@ -206,11 +247,8 @@ def container_add(container_id, ip, interface):
         next_hops[ip.version]
     except KeyError:
         print "This node is not configured for IPv%d." % ip.version
-        sys.exit(1)
-
-    # Assign the IP
-    if not client.assign_address(pool, ip):
-        print "IP address is already assigned in pool %s " % pool
+        # Ignoring return as we don't care if the IP is already unallocated
```

Comment:
Can you expand this comment.  It wasn't immediately clear to me that `get_ip_and_pool()` actually does the IP allocation, so we need to `release_ips()` here to clean up.

Furthermore, I think we do care about the return value of `release_ips()`, at least from a debugging standpoint.  If `ip` is not allocated then something wrong and unexpected has happened.  We should print an error.